lib/rspec/core/formatters.rb

```diff
@@ -194,7 +194,9 @@ def register(formatter, notifications)
 
     def duplicate_formatter_exists?(new_formatter)
       @formatters.any? do |formatter|
-        formatter.class == new_formatter.class && formatter.output == new_formatter.output
+        formatter.class == new_formatter.class &&
+          formatter.respond_to?(:output) == new_formatter.respond_to?(:output) &&
+          (formatter.respond_to?(:output) ? formatter.output == new_formatter.output : true)
```

Comment:
I think I'd prefer we broke this up into something more understandable than a complex boolean, what do you think about leaving the class check, but moving the output check out to a method like this:
```suggestion
        formatter.class == new_formatter.class && has_matching_output?(formatter, new_formatter)
```

This would then enable us to have a private method with a return rather than a compound if like this:
```
        # elsewhere
        private
        def has_matching_output?(formatter, new_formatter)
          return true unless formatter.respond_to?(:output) && new_formatter.respond_to?(:output)
           formatter.output == new_formatter.output
        end
```